File: BinaryTree/php/howHigh.php
<?php

declare(strict_types=1);

require_once __DIR__ . '/TreeNode.php';

function howHigh(?TreeNode $root): int
{
    if ($root === null) {
        return -1;
    }

    return 1 + max(howHigh($root->left), howHigh($root->right));
}

File: BinaryTree/java/howHigh.java
class HowHigh {
  public static int howHigh(TreeNode root) {
    if (root == null) {
      return -1;
    }
    return 1 + Math.max(howHigh(root.left), howHigh(root.right));
  }

  public static void run() {
    // intentionally empty
  }
}
